### PR TITLE
Feature/api docs

### PR DESCRIPTION
**Pull request by starkipraggy**
For documentation

### CODE REVIEW

**Comment by eatedcookie** [comment]
looks fine to me, someone else check and merge?

**Comment by 9gix** [comment]
Please note that this PR includes some changes that is outdated. 
Especially have a look at `DoLahTest/unittest1.cpp`. Why would such changes happened in API Docs Branch? 

I can't approve the merge yet, because I can't understand these changes yet.

**Comment by starkipraggy** [comment]
I don't understand what you don't understand?

Calendar.cpp changes are in line with the changes in master. Remember that Calendar.cpp is behind. New changes are from me rearranging the methods to follow the following pattern: Constructor, destructor, getty, and then execution methods.

Unittest1.cpp was renamed to CalendarUnitTest.cpp in master so it doesn't exist anymore in master but still exists in this branch.

**Comment by 9gix** [comment]
> I don't understand what you don't understand?

It's just a simple diff. All you need to do is spot the difference in these colors
* **RED** -> removed
* **GREEN** -> added

> Calendar.cpp changes are in line with the changes in master. Remember that Calendar.cpp is behind. New changes are from me rearranging the methods to follow the following pattern: Constructor, destructor, getty, and then execution methods.

That's not the issues. Please see the differences to know what I meant. but this is not a big issue. I can still accept this part. However there are other part, I can't accept.

> Unittest1.cpp was renamed to CalendarUnitTest.cpp in master so it doesn't exist anymore in master but still exists in this branch.

Precisely, it is renamed, meaning it shouldn't even be there after merging into master. but why does it still have to exists even after merging it into master? Please see the diff. It's just a simple spot the difference thingy. "RED" and "GREEN". https://github.com/cs2103aug2015-w11-2c/DoLah/pull/76/files

File: DoLah/DoLah.vcxproj.filters [hunk 1]
```diff
@@ -1,27 +1,19 @@
...
-    <ClCompile Include="Commands\DeleteTaskCommand.cpp" />
-    <ClCompile Include="Commands\EditTaskCommand.cpp" />
-    <ClCompile Include="Commands\RedoTaskCommand.cpp" />
-    <ClCompile Include="Commands\SearchTaskCommand.cpp" />
-    <ClCompile Include="Commands\UndoTaskCommand.cpp" />
-    <ClCompile Include="DoLahClient.cpp" />
-    <ClCompile Include="Models\DeadlineTask.cpp" />
-    <ClCompile Include="Models\EventTask.cpp" />
-    <ClCompile Include="Models\Task.cpp" />
-    <ClCompile Include="CommandInvoker.cpp" />
```
Comment (9gix) [inline DoLah/DoLah.vcxproj.filters]: why was this removed??

File: DoLah/DoLah.vcxproj.filters [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-    <ClCompile Include="Models\CommandHistory.cpp" />
```
Comment (9gix) [inline DoLah/DoLah.vcxproj.filters]: why was this removed?

File: DoLah/DoLah.vcxproj.filters [hunk 3]
```diff
@@ -1,27 +1,19 @@
...
-    <ClCompile Include="CommandParser.cpp" />
-    <ClCompile Include="CommandTokenizer.cpp" />
```
Comment (9gix) [inline DoLah/DoLah.vcxproj.filters]: more file being removed here... :-1:
